### ISSUE TITLE
[rancher-gatekeeper-crd] Check for PSP compatibility in target cluster

### ISSUE THREAD

**Issue by stormqueen1990** [rancher/rancher#39913]
Parent ticket: https://github.com/rancher/rancher/issues/39366

Kubernetes 1.25 drops support for PSP, so additional checks need to be added.

Rancher team 3 is following the proposal outlined in "Kubernetes v1.25 PSP to PSA migration proposal document"

Every chart that ships with a PSP must be changed to add a new condition checking for the PSP capability in the target cluster:

**Capabilities check for Helm**

```
{{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
```

This change allows the charts to still work with PSPs in Kubernetes versions prior to v1.25, and work in Kubernetes v1.25 or higher by skipping installation of PSPs.

**Charts that need updating**:

* rancher-gatekeeper-crd

If a cluster role in a chart contains PSP's along with other permissions extra work may be needed to achieve the intended behavior of that role.

**Comment by prachidamle** [comment]
@mitulshah-suse @doflamingo721 See we need a change to the gateleeper-crd chart for the PSP check

**Comment by rishabhmsra** [comment]
Validated on rancher v2.7-head(65089ee)

- Provisioned ec2 node driver, k8s `v1.25.2-rancher1-1` and `v.124.8-rancher1-1` clusters(1-cp, 1-e, 2-w)
  -  Installed OPA gatekeeper `101.1.0+up3.10.0` chart on both downstream clusters.
  - Chart installed successfully.

  ![gatekeeper](https://user-images.githubusercontent.com/36376154/208908555-a1a372a4-753b-4a68-955d-eb6ba6cc3da7.png)
  - To validate the chart, created the constraints using the default templates for Namespace and Repository and it's working as expected.

**Comment by rishabhmsra** [comment]
Re-opening this issue since the upgrade checks are failing:
- On rancher v2.7-head(8062f1e)
- Created aws node driver, k8s v1.24.8-rancher1-1 cluster(1-cp,1-e,2-w)
- Installed OPA gatekeeper chart version `101.0.0+up3.9.0`
- Upgraded the cluster to k8s version v1.25.5-rancher1-1
- Validated the OPA gatekeeper chart by creating the constraints from the default templates `k8sallowedrepos` and `k8srequiredlabels` and the constrains are working as expected.
- However on upgrading the chart to version `101.1.0+up3.10.0` it failed with bellow error:
```
helm upgrade --history-max=5 --install=true --namespace=cattle-gatekeeper-system --timeout=10m0s --values=/home/shell/helm/values-rancher-gatekeeper-crd-101.1.0-up3.10.0.yaml --version=101.1.0+up3.10.0 --wait=true rancher-gatekeeper-crd /home/shell/helm/rancher-gatekeeper-crd-101.1.0-up3.10.0.tgz
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "rancher-gatekeeper-crd-manager" namespace: "cattle-gatekeeper-system" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
ensure CRDs are installed first
```

**Comment by mitulshah-suse** [comment]
Pending on https://github.com/rancher/rancher/pull/40104

**Comment by stormqueen1990** [comment]
I believe this issue has been superseded by #40344.

**Comment by mitulshah-suse** [comment]
Closing this issue since approach for this setup has been changed. Refer to https://github.com/rancher/rancher/issues/40331 for further details
